submarine-sdk/pysubmarine/submarine/store/model_registry/sqlalchemy_store.py

```diff
@@ -0,0 +1,570 @@
+
+    def transition_model_version_stage(self, name: str, version: int, stage: str) -> ModelVersion:
+        """
+        Update model version stage.
+        :param name: Registered model name.
+        :param version: Registered model version.
+        :param stage: New desired stage for this model version.
+        :return: A single :py:class:`submarine.entities.model_registry.ModelVersion` object.
+        """
+        with self.ManagedSessionMaker() as session:
+            last_updated_time = datetime.now()
+
+            sql_model_version = self._get_model_version(session, name, version)
+            sql_model_version.current_stage = get_canonical_stage(stage)
+            sql_model_version.last_updated_time = last_updated_time
+            sql_registered_model = sql_model_version.registered_model
+            sql_registered_model.last_updated_time = last_updated_time
+            self._save_to_db(session, [sql_model_version, sql_registered_model])
+            return sql_model_version.to_submarine_entity()
+
+    def delete_model_version(self, name: str, version: int) -> None:
+        """
+        Delete model version in backend.
+        :param name: Registered model name.
+        :param version: Registered model version.
+        :return: None
+        """
+        with self.ManagedSessionMaker() as session:
+            updated_time = datetime.now()
+            sql_model_version = self._get_model_version(session, name, version)
+            sql_registered_model = sql_model_version.registered_model
+            sql_registered_model.last_updated_time = updated_time
+            session.delete(sql_model_version)
+            self._save_to_db(session, sql_registered_model)
+            session.flush()
+
+    def get_model_version(self, name: str, version: int) -> ModelVersion:
+        """
+        Get the model version instance by name and version.
+        :param name: Registered model name.
+        :param version: Registered model version.
+        :return: A single :py:class:`submarine.entities.model_registry.ModelVersion` object.
+        """
+        with self.ManagedSessionMaker() as session:
+            sql_model_version = self._get_model_version(session, name, version, True)
+            return sql_model_version.to_submarine_entity()
+
+    def list_model_version(self, name: str, filter_tags: List[str] = None) -> List[ModelVersion]:
+        """
+        List of all model versions that satisfy the filter criteria.
+        :param name: Registered model name.
+        :param filter_tags: Filter tags, defaults not to filter any tags.
+        :return: A List of :py:class:`submarine.entities.model_registry.ModelVersion` objects
+                that satisfy the search expressions.
+        """
+        conditions = [SqlModelVersion.name == name]
+        if filter_tags is not None:
+            conditions = [
+                SqlModelVersion.model_tags.any(SqlModelTag.tag.contains(tag)) for tag in filter_tags
```

Comment:
Is it wrong to reassign "conditions" when filter_tags is not None?

Reply:
Thanks for point out this problem